src/psyclone/psyir/nodes/reference.py

```diff
@@ -79,6 +74,20 @@ def symbol(self):
         '''
         return self._symbol
 
+    @symbol.setter
+    def symbol(self, symbol):
+        '''
+        :param symbol: the new symbol being referenced.
+        :type symbol: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        :raises TypeError: if the symbol argument is not of type Symbol.
+
+        '''
+        if not isinstance(symbol, Symbol):
+            raise TypeError("In Reference initialisation expecting a symbol "
+                            "but found '{0}'.".format(type(symbol).__name__))
+        self._symbol = symbol
```

Comment:
Please could you update the error message slightly now that this might not be being called when initialising a Reference.

I was expecting pylint to complain about the fact that self._symbol is not defined in the class constructor but it doesn't. Perhaps it has been made smarter?

Reply:
Updated error message